File: examples/generate-a-pdf/rest/java/GeneratePdf.java
// Generates a PDF from a URL using the Browserless /pdf endpoint.
// java.net.http is included in JDK 11+ — no extra dependency needed.
//
// Run: javac GeneratePdf.java && java GeneratePdf

import java.io.*;
import java.net.URI;
import java.net.http.*;

public class GeneratePdf {
    private static final String TOKEN = "YOUR_API_TOKEN_HERE";

    public static void main(String[] args) throws Exception {
        String endpoint = "https://production-sfo.browserless.io/pdf?token=" + TOKEN;
        String jsonData = """
            {
                "url": "https://example.com",
                "options": {
                    "displayHeaderFooter": true,
                    "printBackground": true,
                    "format": "A4"
                }
            }""";

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint))
            .header("Cache-Control", "no-cache")
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(jsonData))
            .build();

        HttpResponse<InputStream> response = client.send(request, HttpResponse.BodyHandlers.ofInputStream());
        try (FileOutputStream fos = new FileOutputStream("output.pdf")) {
            response.body().transferTo(fos);
        }
        System.out.println("PDF saved as output.pdf.");
    }
}

File: examples/generate-a-pdf/rest/csharp/GeneratePdf.cs
// Generates a PDF from a URL using the Browserless /pdf endpoint.
// Uses System.Net.Http and System.Text.Json from the .NET standard library.
//
// Run: dotnet run

using System.Net.Http;
using System.Text;
using System.Text.Json;

const string Token = "YOUR_API_TOKEN_HERE";
string endpoint = $"https://production-sfo.browserless.io/pdf?token={Token}";

var payload = new
{
    url = "https://example.com",
    options = new { displayHeaderFooter = true, printBackground = true, format = "A4" },
};

using var client = new HttpClient();
var request = new HttpRequestMessage(HttpMethod.Post, endpoint);
request.Headers.Add("Cache-Control", "no-cache");
request.Content = new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json");

var response = await client.SendAsync(request);
var contentStream = await response.Content.ReadAsStreamAsync();
using var fileStream = new FileStream("output.pdf", FileMode.Create, FileAccess.Write);
await contentStream.CopyToAsync(fileStream);
Console.WriteLine("PDF saved as output.pdf.");
